tests/platform_tests/fwutil/test_fwutil.py

```diff
@@ -0,0 +1,105 @@
+import re
+import pytest
+import os
+import json
+
+from fwutil_common import call_fwutil, show_firmware, upload_platform, find_pattern
+
+DEVICES_PATH="/usr/share/sonic/device"
+
+def test_fwutil_show(duthost):
+    """Basic Validation that fwutil show runs without error"""
+    assert show_firmware(duthost)
+
+def test_fwutil_install_file(duthost, localhost, pdu_controller, fw_pkg, random_component):
+    """Tests manually installing firmware to a component from a file."""
+    assert call_fwutil(duthost, 
+            localhost, 
+            pdu_controller, 
+            fw_pkg, 
+            component=random_component, 
+            basepath=os.path.join(DEVICES_PATH, duthost.facts['platform']))
+
+def test_fwutil_install_url(duthost, localhost, pdu_controller, fw_pkg, random_component, host_firmware):
+    """Tests manually installing firmware to a component from a URL."""
+    assert call_fwutil(duthost,
+            localhost,
+            pdu_controller,
+            fw_pkg, 
+            component=random_component, 
+            basepath=host_firmware)
+
+def test_fwutil_install_bad_name(duthost, fw_pkg):
+    """Tests that fwutil install validates component names correctly."""
+    out = duthost.command("fwutil install chassis component BAD fw BAD.pkg", module_ignore_errors=True)
+    pattern = re.compile(r'.*Invalid value for "<component_name>"*.')
+    found = find_pattern(out['stderr_lines'], pattern)
+    assert found
+
+def test_fwutil_install_bad_path(duthost, fw_pkg, random_component):
+    """Tests that fwutil install validates firmware paths correctly."""
+    out = duthost.command("fwutil install chassis component {} fw BAD.pkg".format(random_component), module_ignore_errors=True)
+    pattern = re.compile(r'.*Error: Invalid value for "<fw_path>"*.')
+    found = find_pattern(out['stderr_lines'], pattern)
+    assert found
+
+def test_fwutil_update_current(duthost, localhost, pdu_controller, fw_pkg, random_component):
+    """Tests updating firmware from current image using fwutil update"""
+    assert call_fwutil(duthost,
+            localhost,
+            pdu_controller,
+            fw_pkg, 
+            component=random_component)
+
+def test_fwutil_update_next(duthost, localhost, pdu_controller, fw_pkg, random_component, next_image):
+    """Tests updating firmware from the "next" image using fwutil update"""
+    assert call_fwutil(duthost, 
+            localhost,
+            pdu_controller,
+            fw_pkg, 
+            component=random_component, 
+            next_image=next_image)
+
+def test_fwutil_update_bad_config(duthost, fw_pkg, random_component):
```

Comment:
@alexrallen how about to parameterize this test? With `pytest`, we can dynamically generate a set of invalid config files

Reply:
I see the logic here, however it is more complex than it would seem to do that since the `platform_components.json` requires a valid chassis name and component name. It becomes simpler / more readable to simply duplicate the code.

Reply:
@alexrallen ok. Then it's up to you to decide.